.vsts-ci/releasePhase.yml

```diff
@@ -22,35 +29,56 @@ jobs:
   - pwsh: |
       Write-Host "##vso[task.setvariable variable=ACR_NAME;]$env:ACR_NAME_VAR"
     displayName: 'Enable ACR'
-
-  - pwsh: |
-      $version = '$(releaseTag)' -replace '^v', ''
-      Write-Host "##vso[task.setvariable variable=Version;]$version"
+    condition: and( succeededOrFailed(), ne(variables['Channel'],''))
+
+  - powershell: |
+      switch('$(Channel)') {
+        'stable' {
+          $releaseTag = '$(stableReleaseTag)'
+          $version = '$(stableReleaseTag)' -replace '^v', ''
+        }
+        'preview' {
+          $releaseTag = '$(previewReleaseTag)'
+          $version = '$(previewReleaseTag)' -replace '^v', ''
+        }
+        default {
+          throw "Unknown channel '$(Channel)'"
+        }
+      }
+      $namespace = $releaseTag.ToLowerInvariant()
+      Write-Host "##vso[task.setvariable variable=dockerNamespace;]$namespace"
+      $version = $releaseTag -replace '^v', ''
+      $command = "vso[task.setvariable variable=version]$version"
+      Write-Host $command
+      Write-Host "##$command"
     displayName: 'Set Version'
+    name: get
+    condition: and( succeededOrFailed(), ne(variables['Channel'],''))
 
   - pwsh: |
-      $namespace = '$(releaseTag)'.ToLowerInvariant()
-      Write-Host "##vso[task.setvariable variable=dockerNamespace;]$namespace"
-    displayName: 'Set dockerNamespace'
-
-  - pwsh: 'Get-ChildItem env:'
+      Get-ChildItem env: | Out-String -Width 1000
     displayName: 'Capture Environment'
+    condition: and( succeededOrFailed(), ne(variables['Channel'],''))
 
   - pwsh: 'docker login $(dockerHost) -u $(dockerUserName) -p $(dockerKey)'
     displayName: 'docker login'
+    condition: and( succeededOrFailed(), ne(variables['Channel'],''))
 
   - pwsh: ' az login --service-principal -u $(az_url) -p $(az_key) --tenant $(az_name)'
     displayName: 'az login'
+    condition: and( succeededOrFailed(), ne(variables['Channel'],''))
 
   - pwsh: |
       az account set --subscription $(StorageSubscriptionName)
-      $querystring = az storage account generate-sas --account-name $(StorageAccount) --services b --resource-types o --expiry ((get-date).AddDays(2).ToString("yyyy-MM-dd")) --permission r --https-only | convertfrom-json
+      $querystring = az storage account generate-sas --account-name $(StorageAccount) --services b --resource-types o --expiry ((get-date).AddDays(1).ToString("yyyy-MM-dd")) --permission r --https-only | convertfrom-json
```

Comment:
any reason for changing from 2 to 1?

Reply:
2 days was for initial testing.  1 day is more than enough time.